PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/TelemetryClientProvider.java

```diff
@@ -0,0 +1,72 @@
+package com.microsoft.azure.toolkit.intellij.azure.sdk.buildtool;
+
+
+import com.microsoft.applicationinsights.TelemetryClient;
+import java.io.FileNotFoundException;
+
+import java.util.Timer;
+import org.json.JSONObject;
+import java.io.InputStream;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.util.TimerTask;
+import java.util.stream.Collectors;
+
+/**
+ * This class reads the instrumentation key from the applicationinsights.json file
+ * and returns a TelemetryClient object with the instrumentation key set.
+ * This object is used to send telemetry data to Application Insights.
+ * The trackMemoryUsage method is used to track the memory usage of the application.
+ * The start method is used to start a timer that runs the trackMemoryUsage method every 10 seconds.
+ */
+public class TelemetryClientProvider {
+
+    public static TelemetryClient getTelemetryClient() {
+
+        TelemetryClient telemetry;
+        telemetry = new TelemetryClient();
+
+        String instrumentationKey = null;
+
+        try {
+            InputStream inputStream = ConnectionStringCheck.class.getClassLoader().getResourceAsStream("META-INF/applicationinsights.json");
+            if (inputStream == null) {
+                throw new FileNotFoundException("File not found on classpath");
+            }
+
+            BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream));
+            String json = reader.lines().collect(Collectors.joining("\n"));
+            reader.close();
+
+            JSONObject jsonObject = new JSONObject(json);
+            instrumentationKey = jsonObject.getString("instrumentationKey");
```

Comment:
Do we need to read this entire file end to end and hold its lines in memory?  What about streaming? IIRC our azure-json library allows for input streams.

Reply:
will look into this:)